LEAF_Request_Portal/sources/Form.php

```diff
@@ -335,140 +335,148 @@ public function getForm($recordID, $limitCategory = null)
     // Expects POST input: $_POST['service'], title, priority, num(categoryID)
     public function newForm($userID)
     {
-        if ($_POST['CSRFToken'] != $_SESSION['CSRFToken'])
-        {
+        if ($_POST['CSRFToken'] != $_SESSION['CSRFToken']) {
             return 'Error: Invalid token.';
         }
+
         $title = XSSHelpers::sanitizeHTML($_POST['title']);
         $_POST['title'] = $title == '' ? '[blank]' : $title;
         $_POST['service'] = !isset($_POST['service']) || $_POST['service'] == '' ? 0 : (int)$_POST['service'];
         $_POST['priority'] = !isset($_POST['priority']) || $_POST['priority'] == '' ? 0 : (int)$_POST['priority'];
 
         if ($_POST['title'] == ''
             || !is_numeric($_POST['service'])
-            || !is_numeric($_POST['priority']))
-        {
+            || !is_numeric($_POST['priority'])
+        ) {
             return 'Error: Please check the data you\'ve entered, and try again.';
         }
 
         $keys = array_keys($_POST);
 
-        $categoryIDsArr = array();
-        if (isset($_POST['title']))
-        {
-            foreach ($keys as $key)
-            {
-                if (strpos($key, 'num') === 0)
-                {
+        $categoryIDsArr = [];
+
+        if (isset($_POST['title'])) {
+            foreach ($keys as $key) {
+                if (strpos($key, 'num') === 0) {
                     $categoryIDsArr[] = XSSHelpers::xscrub(strtolower(substr($key, 3)));
                 }
             }
         }
-        if (count($categoryIDsArr) === 0)
-        {
+
+        if (count($categoryIDsArr) === 0) {
             return 'Error: No forms selected. Please Select a form and try again.';
         }
 
         //check category visible status.  unpublished forms (visibility -1) should not be created.
-        $pubVars = array(
-            ":categoryIDs" => implode(',', $categoryIDsArr)
-        );
-        $pubSQL = "SELECT * FROM categories WHERE FIND_IN_SET(categoryID, :categoryIDs) AND `visible` = -1";
-        $unpublishedForms = $this->db->prepared_query($pubSQL, $pubVars);
+        $placeholders = implode(',', array_fill(0, count($categoryIDsArr), '?'));
+
+
+        $sql = "SELECT *
```

Comment:
This looks like it's just getting a count?  So doesn't need a select star.  And can just return a count(*).